docs/docs/integrations/tools/bing_search.ipynb

```diff
@@ -143,34 +145,179 @@
+      "\n",
+      "\u001b[1m> Finished chain.\u001b[0m\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'input': 'What happened in the latest burning man floods?',\n",
+       " 'output': \"The latest Burning Man festival experienced heavy rain and flooding, which resulted in thousands of festivalgoers being stranded. Some attendees had to hike for miles to safety. The rain caused flash flooding in parts of the festival site, including the Black Rock Forest where the event takes place. Campsites were flooded, and thick mud made movement difficult. Eventually, after the intense rain over the weekend, attendees were able to make a mass exodus from the festival. The Pershing County Sheriff's Office warned festivalgoers about the flooding and encouraged them to leave for safety.\"}"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import getpass\n",
+    "import os\n",
+    "\n",
+    "from langchain import hub\n",
+    "from langchain.agents import AgentExecutor, create_openai_functions_agent\n",
+    "from langchain_openai import AzureChatOpenAI\n",
+    "\n",
+    "os.environ[\"AZURE_OPENAI_API_KEY\"] = getpass.getpass()\n",
+    "os.environ[\"AZURE_OPENAI_ENDPOINT\"] = \"https://<your-endpoint>.openai.azure.com/\"\n",
+    "os.environ[\"AZURE_OPENAI_API_VERSION\"] = \"2023-06-01-preview\"\n",
+    "os.environ[\"AZURE_OPENAI_DEPLOYMENT_NAME\"] = \"<your-deployment-name>\"\n",
+    "\n",
+    "instructions = \"\"\"You are an assistant.\"\"\"\n",
+    "base_prompt = hub.pull(\"langchain-ai/openai-functions-template\")\n",
```

Comment:
Suggest using [tool calling](https://python.langchain.com/v0.1/docs/use_cases/tool_use/agents/) instead of function calling as I believe function calling is deprecated by OpenAI.

Reply:
Created this PR as a follow-up.
https://github.com/langchain-ai/langchain/pull/23184